src/antsibull_core/subprocess_util.py

```diff
@@ -28,16 +29,33 @@
 async def _stream_log(
     name: str, callback: Callable[[str], Any] | None, stream: asyncio.StreamReader
 ) -> str:
-    line = await stream.readline()
+    # We do not simply use stream.readline() since it has a line length limit.
+    # While we set this limit already to 8 MB (the default is 64 KB), we still
+    # want to cover longer lines as well, so we use stream.readuntil('\n')
+    # and manually handle the case of longer lines.
     lines = []
+    line_parts = []
+    sep = b'\n'
     while True:
+        try:
+            line_parts.append(await stream.readuntil(sep))
+        except IncompleteReadError as e:
+            line_parts.append(e.partial)
+        except LimitOverrunError as e:
+            part = await stream.read(e.consumed)
+            line_parts.append(part)
+            if part and not part.endswith(sep):
```

Comment:
I added a small extra test (`part.endswith(sep)`). This should not happen except if asyncio changes its behavior. I'm unsure whether this should really be here. @gotmax23 what do you think?

Reply:
I don't think so. It would be nice if the documentation of that exception's `consumed` value would be a bit clearer (https://docs.python.org/3/library/asyncio-stream.html#asyncio.StreamReader.readuntil) though to clearly state that the range it describes does not include the separator.